src/core/EnvelopeAndLfoParameters.cpp

```diff
@@ -402,6 +406,8 @@ void EnvelopeAndLfoParameters::loadSettings( const QDomElement & _this )
 
 void EnvelopeAndLfoParameters::updateSampleVars()
 {
+	m_paramMutex.lock();
```

Comment:
Please use a [QMutexLocker](https://doc.qt.io/qt-4.8/qmutexlocker.html) here to ensure safe unlocking in cases of exceptions, etc.